org.eclipse.xtext.ide.tests/src/org/eclipse/xtext/ide/tests/server/OpenDocumentTest.xtend

```diff
@@ -132,4 +132,15 @@ class OpenDocumentTest extends AbstractTestLangLanguageServerTest {
         close(fileURI);
         assertFalse(diagnostics.get(referencingFileURI).empty)
     }
+
+    @Test
+    def void testInMemory_GH_456() {
+        initialize
+        val uri = 'inmemory:/foo/bar.testlang';
+        open(uri, '''
+            type Foo { }
+        ''');
+        assertNotNull('''Diagnostics could be empty but never null.''', diagnostics.get(uri));
```

Comment:
Just in case you make any additional modification in the PR, could you please fix my misleading assertion message. That is eventually not true. It can be null, but not in this case.